src/Examples/Sorting/Sequential/InsertionSort.agda

```diff
@@ -113,37 +113,64 @@ sort/correct (x ∷ xs) u =
   ) , sorted-x∷xs'
 
 sort/cost : cmp (Π (list A) λ _ → meta ℂ)
-sort/cost l = length l ²
+sort/cost l = λ _ → length l ²
+
+Modal : (⋄ : tp pos → tp pos) (A : tp pos) → Set
+Modal ⋄ A = val (⋄ A) ↔ val A
+
+postulate
+  lemma : (A : tp pos) (e : cmp (F A)) (v : val A) → ◯ (e ≡ ret v) →
+    (X : tp neg) (f : val A → cmp X) →
+    bind X e f ≡ bind X e (const (f v))
```

Comment:
Verify: is this true semantically?

Reply:
This is surely false semantically; you can use it to show that `step` is the identity without any assumptions.

Reply:
Hmm - how so? (Is this still the case if `A` is open-modal?)

Reply:
Even if `A` is open-modal it is not valid. Remember that `F A` is interpreted as something like `Cl Cost * A` or something along these lines. Let `X = F A` and let `f = ret`. Suppose for instance that `A = Unit`, which is open-modal.

```
  lemma Unit (step; ret tt) tt refl (F A) ret : (bind (step; ret tt) ret) == (bind (F A) (ret tt) ret)
```

The `refl` witnesses that `Opn (step; ret tt == ret tt)`. Computing further from the application of `lemma` using the unit laws of the monad, we evidently have `step; ret tt == ret tt`.

Reply:
Ah, I have made a mistake in the deduction here (as discussed in chat). I believe this might be consistent with the counting model if `A` is open-modal, but not otherwise.

For example, suppose that `A` is `Cl Nat`. Then we deduce `bind (ret (eta 5)) ret == bind (ret (eta 5)) (\_ -> eta 6)` but it happens this implies that `eta 5 == eta 6`.

Reply:
(Resolved offline; this axiom is okay as long as `A` is open-modal.)